### PR TITLE
Add hooks to UI events

### LINKED ISSUE #36: Get profile data during the first login
Currently after getting the `User` object from a successful login in `PasswordActivity` we need to make another call to get the user data. This means we're in this unconfirmed state between logging in and receiving any useful user data to display. The call itself (to fetch user details) can also fail which means we'll stay with an empty description of the user.

It would be nice if the object retrieved during the login flow contained full `ProfileDetails` that we could update when necessary.

Because here's the current flow:
- user opens up the `PasswordActivity`
- user passes his credentials
- user is *waiting* for logging in
- user returns to the actual app activity where info about him should be displayed but he needs to wait for that (again, with uncertain state)

I believe this flow could be better from the user experience side.

### PR DESCRIPTION
**Pull request by Nexcius**
Proposal to fix  #36

@aliak00 @Astach What do you think? This is just a hook. Would it make sense to be able to show a UI error or something? How do you handle that, Ali?

### CODE REVIEW

File: ui/src/main/java/com/schibsted/account/ui/UiHoks.kt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package com.schibsted.account.ui
+
+import com.schibsted.account.session.User
+
+interface UiHoks {
+    class OnReadyListener(private val block: () -> Unit) {
```
[the remark this answers is not in the document]
Comment (Nexcius) [reply ui/src/main/java/com/schibsted/account/ui/UiHoks.kt]: Changed to `OnProceedListener` should be generic

File: ui/src/main/java/com/schibsted/account/ui/UiHoks.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        fun onReady() {
```
[the remark this answers is not in the document]
Comment (Nexcius) [reply ui/src/main/java/com/schibsted/account/ui/UiHoks.kt]: Changed to `proceed`

File: example/src/main/java/com/schibsted/account/example/App.java [hunk 3]
```diff
@@ -4,13 +4,19 @@
...
+import com.schibsted.account.common.util.Logger;
 import com.schibsted.account.session.User;
 import com.schibsted.account.ui.AccountUiHook;
 
 public class App extends Application implements AccountUiHook {
     @Override
-    public void onUiClosing(@NonNull User user, @NonNull OnProceedListener onProceedListener) {
-        Log.d("XXX", "IU can see the UIs are closing! " + user.getUserId().getId());
+    public void onLoginCompleted(@NonNull User user, @NonNull OnProceedListener onProceedListener) {
+        Logger.debug("XXX", "IU can see the UIs are closing! " + user.getUserId().getId());
```
Comment (Astach) [inline example/src/main/java/com/schibsted/account/example/App.java]: This log isn't really needed

Comment (Nexcius) [reply example/src/main/java/com/schibsted/account/example/App.java]: this is just to provide clarity on how it works

**Pull request merged**